lib/YaoSubspaceArrayReg/src/type.jl

```diff
@@ -46,40 +48,40 @@ Create a `SubspaceArrayReg` in zero state in given subspace.
 - `n`: required, number of atoms (qubits).
 - `subspace`: required, the subspace of rydberg state.
 """
-YaoArrayRegister.zero_state(subspace::Subspace) = zero_state(ComplexF64, subspace)
+YaoArrayRegister.zero_state(subspace::Subspace; nlevel=2) = zero_state(ComplexF64, subspace; nlevel)
 
-function YaoArrayRegister.zero_state(::Type{T}, s::Subspace) where {T}
+function YaoArrayRegister.zero_state(::Type{T}, s::Subspace; nlevel=2) where {T}
     state = zeros(T, length(s))
     state[1] = 1
-    return SubspaceArrayReg(state, s)
+    return SubspaceArrayReg{nlevel}(state, s)
 end
 
 """
-    rand_state(subspace)
+    rand_state([T=ComplexF64], subspace; nlevel=2)
 
 Create a random state in the given subspace.
 """
 YaoArrayRegister.rand_state(s::Subspace) = rand_state(ComplexF64, s)
 
-function YaoArrayRegister.rand_state(::Type{T}, s::Subspace) where {T<:Complex}
+function YaoArrayRegister.rand_state(::Type{T}, s::Subspace; nlevel=2) where {T<:Complex}
     state = normalize!(rand(T, length(s)))
```

Comment:
if you want to support `nlevel` the subspace object should also be updated?

Reply:
subspace now use integers rather than dit string, it should be refactored a bit. Let us do it later.